plugins/inputs/traefik/traefik.go

```diff
@@ -0,0 +1,138 @@
+package traefik
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Traefik struct {
+	Address                      string
+	ResponseTimeout              internal.Duration
+	IncludeStatusCodeMeasurement bool
+	lastRequestTiming            float64
+}
+
+type HealthCheck struct {
+	UptimeSec              float64   `json:"uptime_sec"`
+	Unixtime               int64     `json:"unixtime"`
+	TotalStatusCodeCount   HttpCodes `json:"total_status_code_count"`
+	TotalCount             int       `json:"total_count"`
+	TotalResponseTimeSec   float64   `json:"total_response_time_sec"`
+	AverageResponseTimeSec float64   `json:"average_response_time_sec"`
+}
+
+type HttpCodes map[string]int
+
+var sampleConfig = `
+# Required Traefik server address, host and port (default: "127.0.0.1")
```

Comment:
Great question. I took over this pull request to help drive it through completion. Adding credentials was not part of the original PR and not really needed to get through the initial completion of this PR. Obviously there are people out there who may want to secure this endpoint, so that functionality should probably be added. However, since the most common use case for this kind of metric data is probably unauthenticated, it makes sense to push this through as is and submit another PR with the added functionality. 

Would you want to work on adding the ability to add credentials?

Reply:
Traefik supports several methods for authentication: https://docs.traefik.io/configuration/entrypoints/.  If we add support for this (and I agree it does not need to be in the initial PR) we should do it in a way that we could configure any of the methods even if we only do HTTP Basic Auth initially.

So, if you do work on this don't use the userinfo section of the URL, instead add options like https://github.com/influxdata/telegraf/blob/master/plugins/inputs/http/http.go#L58-L60

@mkboudreau  One thing I would like to add initially though is support for TLS.  Should be pretty easy we just need to mirror the code in the http input: https://github.com/influxdata/telegraf/blob/master/plugins/inputs/apache/apache.go#L100-L115

Reply:
@mkboudreau wrote
> Would you want to work on adding the ability to add credentials?
sorry, i dont know GO, so i think i would not be helpfull.
i tried to read the mongodb input plugin code to see how they strip credentials from the server... was not able to understand it at all ;-)

Reply:
@jkowalleck Are you interested in HTTP Basic Auth then?  I could add this in once it's merged pretty easily.

Reply:
@danielnelson  Since traefik supports BASIC and DIGEST i suggest having both available.
see https://docs.traefik.io/configuration/entrypoints/

my current setup uses BASIC auth.
currently using `httpjson` input plugin as following:
```toml
[[inputs.httpjson]] ## traefik
#   ## NOTE This plugin only reads numerical measurements, strings and booleans
#   ## will be ignored.

#   ## Name for the service being polled.  Will be appended to the name of the
#   ## measurement e.g. httpjson_webserver_stats
#   ##
#   ## Deprecated (1.3.0): Use name_override, name_suffix, name_prefix instead.
#   name = "webserver_stats"
  name_override = "traefik"  # overwrite target - default target is "httpjson"

#   ## URL of each server in the service's cluster
#   ## !!!! DO NOT ADD auth here - it might(will) be written to the (influx) output plugin as a tag
  servers = ["http://traefik:57475/health"]

#   ## Set response_timeout (default 5 seconds)
#   response_timeout = "5s"

#   ## HTTP method to use: GET or POST (case-sensitive)
#   method = "GET"

#   ## HTTP Headers (all values must be strings)
  [inputs.httpjson.headers]
    authorization = "Basic c3RhdHM6c3RhdHM="
```

Reply:
@jkowalleck I'll add support for Basic Auth after we merge, might take a bit longer to get digest.  It will probably look more like the `http` input (successor to httpjson in 1.6).

Reply:
> @mkboudreau One thing I would like to add initially though is support for TLS. Should be pretty easy we just need to mirror the code in the http input: https://github.com/influxdata/telegraf/blob/master/plugins/inputs/apache/apache.go#L100-L115

@danielnelson I should be able to update and add TLS support. Might have to wait until at least next week.